lib/model/channel.dart

```diff
@@ -153,9 +153,8 @@ mixin ChannelStore on UserStore {
       case ChannelPostPolicy.any:             return true;
       case ChannelPostPolicy.fullMembers:     {
         if (!role.isAtLeast(UserRole.member)) return false;
-        return role == UserRole.member
-          ? hasPassedWaitingPeriod(user, byDate: byDate)
-          : true;
+        return role != UserRole.member
+          || hasPassedWaitingPeriod(user, byDate: byDate);
```

Comment:
The new code here seems more confusing than the old. I feel like to understand it, I have to unpack it back into basically what we have now.

So this looks like an overzealous lint rule to me. But probably easier to accommodate it than to ignore it.

How about:
```dart
        if (!role.isAtLeast(UserRole.member)) return false;
        if (role == UserRole.member) {
          return hasPassedWaitingPeriod(user, byDate: byDate);
        }
        return true;
```